src/main/java/uk/gov/hmcts/reform/preapi/media/MediaKind.java

```diff
@@ -58,8 +62,62 @@ public MediaKind(
     }
 
     @Override
-    public String playAsset(String assetId) {
-        throw new UnsupportedOperationException();
+    public PlaybackDTO playAsset(String assetName, String userId) {
+        if (getAsset(assetName) == null) {
+            throw new NotFoundException("Asset: " + assetName);
+        }
+
+        var locators = mediaKindClient.getAssetStreamingLocators(assetName);
+
+        var locator = locators.getStreamingLocators()
+            .stream()
+            .filter(l -> l.getName().equals(userId))
+            .findFirst()
+            .orElse(null);
+
+        if (locator == null) {
+            var properties = MkCreateStreamingLocator.MkCreateStreamingLocatorProperties.builder()
+                .assetName(assetName)
+                .streamingPolicyName("Predefined_ClearStreamingOnly")
+                .endTime(Instant.now().plusSeconds(3600).toString())
+                .build();
+            try {
+                mediaKindClient.putStreamingLocator(userId, MkCreateStreamingLocator.builder()
+                    .properties(properties).build());
+            } catch (FeignException e) {
+                throw new MediaKindException();
+            }
+        }
+
+        // TODO check streaming endpoint running + start if not
+        String hostName;
+        try {
+            hostName = mediaKindClient.getStreamingEndpointByName("default").getProperties().getHostName();
+        } catch (FeignException.NotFound e) {
+            throw new NotFoundException("Streaming Endpoint: default");
+        } catch (FeignException e) {
+            throw new MediaKindException();
+        }
```

Comment:
Let everything but the 404 bubble up
```suggestion
        }
```